### PR TITLE
Contextual Menu empty submenu changes

### PR DESCRIPTION
**Pull request by kelseyyoung**
Contextual Menu submenus that have no items will have the chevron icon and call onMenuOpened

- Check only for the existence of a submenu, not that it has items
- Apply changes to CommandBar
- Add tests for icon check and onMenuOpened on click

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ x] Include a change request file using `$ npm run change`

#### Description of changes

Previously, an empty submenu in a Contextual Menu would be treated as a normal button. However, we need an empty submenu to behave like one that has items to allow for dynamically loading menus (populated on an onMenuOpened call). We now only check that a submenu has been defined ('subMenuProps' is defined, or the soon to be deprecated 'items' property is defined) to render the chevron icon and have the submenu call onMenuOpened

#### Focus areas to test

Added tests for Contextual Menu and Command Bar, which creates Contextual Menu items

### CODE REVIEW

**Comment by micahgodbolt** [comment]
I feel like a broken record, but with the new commandBar coming out in 6.0 (hopefully this month), most of this code is going to be going away soon. Can you confirm whether or not this works properly with the commandBar currently in experiments? If it does, I'm find with this fix. If it does not, we need to make the change there so that you aren't broken as soon as you upgrade to 6.0.

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 1]
```diff
@@ -47,10 +47,8 @@ export interface IContextualMenuState {
   submenuDirection?: DirectionalHint;
 }
 
-export function hasSubmenuItems(item: IContextualMenuItem) {
-  let submenuItems = getSubmenuItems(item);
-
-  return !!(submenuItems && submenuItems.length);
+export function hasSubmenu(item: IContextualMenuItem) {
+  return item.subMenuProps ? !!item.subMenuProps : !!item.items;
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: can you simplify this to 
``` typescript
!!(item.subMenuProps || item.items)?
```

**Comment by christiango** [comment]
Thanks for the heads up @micahgodbolt !

**Comment by kelseyyoung** [comment]
@micahgodbolt I tested in experiments and CommandBar works as expected

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx (hunk 1 above)
Comment (kelseyyoung) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: sure

**Pull request merged**